src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ToolEnvironment.java

```diff
@@ -181,6 +182,8 @@ void setElementToTreePath(Element e, TreePath tree) {
     }
 
     public Kind getFileKind(TypeElement te) {
+        if (te.asType().getKind() == TypeKind.ERROR)
+            return Kind.OTHER;
```

Comment:
On a medium deep-dive, I looked at whether this was the "best" solution, or whether it was "second best" and that we should fix the origin of the `ClassCastException` in `Symbol.outermostclass()`. Bottom line: while we could fix it for this case, in general, there is no way to guarantee that there is a reasonable result of type `ClassSymbol` for every reasonable symbol for which this method is called.  Arguably, `.outermostclass` could do more checking and/or throw some exception and/or return `null`, but that is not the javac style, and we would still have to deal with the issue at call sites, like this one.

It's also debatable whether the correct return value is `OTHER` or `null`. `OTHER` has typically meant _some kind of file with an unknown extension_ and not _no file_.  The spec is unclear on this point, and clarifying the spec is out of scope in this PR, and would require a CSR.  So, for now, `OTHER` is OK.